app/features-json/repos_json_controller.rb

```diff
@@ -16,5 +17,57 @@ class RepositoryJSONController < AuthenticatedControllerBase
 
       return all_repos_view_models.to_json
     end
+
+    # Accepts the repo full name and branch as query params
+    # Ex. "../repos/lanes?repo_full_name=nakhbari%2FHelloWorld&branch=master"
+    get "#{HOME}/lanes" do
+      repo_full_name = params[:repo_full_name]
+      branch = params[:branch]
+
+      provider_credential = check_and_get_provider_credential(
+        type: FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github]
+      )
+
+      github_service = FastlaneCI::GitHubService.new(provider_credential: provider_credential)
+
+      selected_repo = github_service.repos.detect do |repo|
+        repo_full_name == repo[:full_name]
+      end
+
+      if selected_repo.nil?
+        raise "Could not find repo, check that your github token has access to repo: #{repo_full_name}}"
+      end
+
+      fastfile_peeker = FastlaneCI::FastfilePeeker.new(
+        provider_credential: provider_credential,
+        notification_service: FastlaneCI::Services.notification_service
+      )
+      repo_config = GitHubRepoConfig.from_octokit_repo!(repo: selected_repo)
+
+      fastfile_parser = fastfile_peeker.fastfile(
+        repo_config: repo_config,
+        sha_or_branch: branch
+      )
+
+      return fetch_available_lanes(fastfile_parser).to_json
+    end
+
+    def fetch_available_lanes(fastfile_parser)
+      # we don't want to show `_before_all_block_`, `_after_all_block_` and `_error_block_`
+      # or a private lane as an available lane
+      lanes = []
+      fastfile_parser.tree.each do |platform, value|
```

Comment:
We already have that code in `project_controller.rb` - the plan is to remove it from project controller as it's only for the `erb`, right?

Reply:
yes. I'm imagining the ERB controllers as a scratch pad and the final code goes here.